SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -600,6 +604,24 @@ void USpatialNetDriver::QueryGSMToLoadMap()
 	GlobalStateManager->QueryGSM(QueryDelegate);
 }
 
+void USpatialNetDriver::OnActorSpawned(AActor* Actor)
+{
+	if (!Actor->GetIsReplicated() ||
+		Actor->GetLocalRole() != ROLE_Authority ||
+		Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_Singleton | SPATIALCLASS_NotSpatialType) ||
+		USpatialStatics::IsActorGroupOwnerForActor(Actor))
+	{
+		// We only want to delete actors which are replicated and we somehow gain local authority over, while not the actor group owner
+		return;
+	}
+
+	FString WorkerType = GetGameInstance()->GetSpatialWorkerType().ToString();
+	UE_LOG(LogSpatialOSNetDriver, Error, TEXT("Worker %s spawned replicated actor %s (owner: %s) but is not actor group owner for actor group %s. The actor will be destroyed in 0.01s"),
+		*WorkerType, *GetNameSafe(Actor), *GetNameSafe(Actor->GetOwner()), *USpatialStatics::GetActorGroupForActor(Actor).ToString());
+	Actor->TearOff();
```

Comment:
Any reason you went for this instead of setting bReplicated to false?

Reply:
Spoke offline, added a comment explaining this